clouddriver-cf/src/main/groovy/com/netflix/spinnaker/clouddriver/cf/cache/Keys.groovy

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2016 Pivotal, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.cf.cache
+
+import com.netflix.frigga.Names
+import com.netflix.spinnaker.clouddriver.cf.CloudFoundryCloudProvider
+import groovy.transform.CompileStatic
+
+/**
+ * @author Greg Turnquist
+ */
+@CompileStatic
+class Keys {
+
+  static enum Namespace {
+    INSTANCES,
+    SERVER_GROUPS,
+    CLUSTERS,
+    APPLICATIONS,
+    LOAD_BALANCERS,
+    ON_DEMAND
+
+    final String ns
+
+    private Namespace() {
+      def parts = name().split('_')
+
+      ns = parts.tail().inject(new StringBuilder(parts.head().toLowerCase())) { val, next -> val.append(next.charAt(0)).append(next.substring(1).toLowerCase()) }
+    }
+
+    String toString() {
+      ns
+    }
+  }
+
+  static Map<String, String> parse(String key) {
+    def parts = key.split(':')
+
+    if (parts.length < 2) {
+      return null
+    }
+
+    def result = [provider: parts[0], type: parts[1]]
+
+    switch (result.type) {
```

Comment:
We have this: https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/cache/Keys.groovy#L54-L56